VenafiPS/Public/Add-TppAdaptableHash.ps1

```diff
@@ -0,0 +1,175 @@
+
+    .EXAMPLE
+    Add-TppAdaptableHash -Path $Path -FilePath 'C:\Program Files\Venafi\Scripts\AdaptableLog\Generic-LogDriver.ps1'
+
+    Update the hash on an adaptable log object.
+
+    .LINK
+    http://VenafiPS.readthedocs.io/en/latest/functions/Add-TppAdaptableHash/
+
+    .LINK
+    https://github.com/Venafi/VenafiPS/blob/main/VenafiPS/Public/Add-TppAdaptableHash.ps1
+
+    .LINK
+    https://docs.venafi.com/Docs/current/TopNav/Content/SDK/WebSDK/r-SDK-POST-Secretstore-add.php
+
+    .LINK
+    https://docs.venafi.com/Docs/currentSDK/TopNav/Content/SDK/WebSDK/r-SDK-POST-Secretstore-ownerdelete.php
+
+    .LINK
+    https://docs.venafi.com/Docs/current/TopNav/Content/SDK/WebSDK/r-SDK-POST-Secretstore-retrieve.php
+    #>
+
+    [CmdletBinding(SupportsShouldProcess)]
+    param (
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript( {
+                if ( $_ | Test-TppDnPath ) {
+                    $true
+                } else {
+                    throw "'$_' is not a valid DN path"
+                }
+            })]
+        [Alias('DN')]
+        [String] $Path,
+
+        [Parameter()]
+        [ValidateSet('Adaptable App')]
+        [ValidateNotNullOrEmpty()]
+        [string] $Class,
+
+        [Parameter()]
+		[string] $Keyname = "Software:Default",
+
+        [Parameter(Mandatory)]
+        [Alias('File')]
+        [string] $FilePath,
+
+        [Parameter()]
+        [psobject] $VenafiSession = $VenafiSession
+    )
+
+    begin {
+        Test-VenafiSession -VenafiSession $VenafiSession -Platform 'TPP'
+
+        $params = @{
+            VenafiSession = $VenafiSession
+            Method        = 'Post'
+        }
+
+        if ( $Class ) {
+            $retrieveVaultID = ( Get-TppAttribute -Path $Path -Class $Class -Attribute 'PowerShell Script Hash Vault Id' ).'PowerShell Script Hash Vault Id'
+        } else {
+            $retrieveVaultID = ( Get-TppAttribute -Path $Path -Attribute 'PowerShell Script Hash Vault Id' ).'PowerShell Script Hash Vault Id'
+        }
+
+        $bytes = [Text.Encoding]::UTF32.GetBytes([IO.File]::ReadAllText($FilePath))
+        $hash = Get-FileHash -InputStream ([System.IO.MemoryStream]::New($bytes))
+        $base64data = [Convert]::ToBase64String([Text.Encoding]::UTF8.GetBytes($hash.hash.ToLower()))
+
+    }
+
+    process {
+        if ( -not $PSCmdlet.ShouldProcess($Path) ) {
+            continue
+        }
+
+        if ( $retrieveVaultID ) {
+            $paramsretrieve = $params
+            $paramsretrieve.UriLeaf = 'SecretStore/retrieve'
+            $paramsretrieve.Body = @{
+                VaultID = $retrieveVaultID
+            }
+
+            $retrieveResponse = Invoke-VenafiRestMethod @paramsretrieve
+
+            if ( $retrieveResponse.Result -ne [TppSecretStoreResult]::Success ) {
+                Write-Error ("Error retrieving VaultID: {0}" -f [enum]::GetName([TppSecretStoreResult], $retrieveResponse.Result))
```

Comment:
in that case, a vault id has been assigned, but there's no value?  that seems odd.  either way, if it's expected it shouldn't be an error

Reply:
To clarify, as I didn't word this that well, Get-TPPAttribute won't find a VaultID if it's new and then this won't be run.
I think if it has a VaultID and for some reason it can't retrieve the hash, then we should stop.